jest/preset.json

```diff
@@ -0,0 +1,26 @@
+{
+  "haste": {
+    "defaultPlatform": "ios",
+    "platforms": ["android", "ios", "native"],
+    "providesModuleNodeModules": [
+      "react-native"
+    ]
+  },
+  "moduleNameMapper": {
+    "^React$": "<rootDir>/node_modules/react"
+  },
+  "modulePathIgnorePatterns": [
+    "<rootDir>/node_modules/react-native/Libraries/react-native/"
+  ],
+  "transform": {
+    "^.+\\.js$": "babel-jest",
+    "^.+\\.(bmp|gif|jpg|jpeg|mp4|png|psd|svg|webp)$": "<rootDir>/node_modules/react-native/jest/assetFileTransformer.js"
+  },
+  "transformIgnorePatterns": [
+    "node_modules/(?!(jest-)?react-native|react-clone-referenced-element)"
+  ],
+  "setupFiles": [
+    "<rootDir>/jest/setup.js"
+  ],
```

Comment:
Do we need all of these options for the subset we're planning to test? It would be nice to trim it down as much as we can.

Reply:
This modified from https://github.com/facebook/react-native/blob/master/jest-preset.json. The version on RN master wouldn't work with the version from NPM and the current RN master isn't on NPM.